src/univariate/continuous/uniform.jl

```diff
@@ -151,7 +152,7 @@ Base.:*(c::Real, d::Uniform) = Uniform(minmax(c * d.a, c * d.b)...)
 
 #### Sampling
 
-rand(rng::AbstractRNG, d::Uniform) = d.a + (d.b - d.a) * rand(rng)
+rand(rng::AbstractRNG, d::Uniform{T}) where T = d.a + d.len * rand(T, rng)
```

Comment:
```suggestion
rand(rng::AbstractRNG, d::Uniform{T}) where T = d.a + d.len * rand(float(T), rng)
```
You need to promote `T` to a floating-point type.

Reply:
Thanks, I missed that one.